modules/fuse/test/fuse_test.py

```diff
@@ -0,0 +1,187 @@
+        Type.DOUBLE:lambda x:np.random.uniform(low=0, high=1000, size=x),
+        Type.STRING:lambda x:generate_string_array(x),
+ 
+    }
+    return pa.array(f[type](length))
+    
+
+
+def assert_dataframe(stored_df:pd.DataFrame, extracted_df:pa.Table):
+    pdf = pa.Table.from_pandas(stored_df)
+    assert extracted_df.equals(pdf),"data frame unmatch"
+def assert_array(stored_arr:pa.Array, extracted_array:pa.Array):
+    assert stored_arr.equals(extracted_array), "array unmatch"
+
+def read_data_from_fuse(vid):
+    with open(os.path.join(test_mount_dir,vid), 'rb') as source:
+        with pa.ipc.open_stream(source) as reader:
+            data = reader.read_all()
+            return data
+
+def test_fuse_array(data,client):
+    id = client.put(data)
+    extracted_data = read_data_from_fuse(str(id)[11:28]+".arrow")
+    print("data: ")
+    print(data)
+    print("extracted data: ")
+    extracted_data = extracted_data.column("a").chunk(0)
+    print(extracted_data)
+    assert_array(data,extracted_data)
+
+def test_fuse_string_array(data,client):
+    id = client.put(data)
+    extracted_data = read_data_from_fuse(str(id)[11:28]+".arrow")
+    print("data: ")
+    print(data)
+    print("extracted data: ")
+    extracted_data = extracted_data.column("a").chunk(0)
+    print(extracted_data)
+    assert compare_two_string_array(data,extracted_data), "string array not the same"
+
+def compare_two_string_array(arr_str_1,arr_str_2):
+    a = arr_str_1
+    b = arr_str_2
+    if(len(a) != len(b)):
+        return False
+    else:
+        for i,j in zip(a,b):
+            if str(i)!= str(j):
+                print("they are different")
+                print(i)
+                print(j)
+                return False
+    return True
+
+def test_fuse_df(data,client):
+    id = client.put(data)
+    extracted_data = read_data_from_fuse(str(id)[11:28]+".arrow")
+    assert_dataframe(data,extracted_data)
+# string_array = generate_array(Type.STRING)
+# int_array =generate_array(Type.INT64)
+# print(int_array)
+if __name__ == "__main__":
+    
+    # logger.basicConfig(filename='test.log', level=logger.DEBUG)
+    print("started")
+
+    # vineyard_server = start_vineyard_server()
+    vineyard_server = asyncio.run(start_vineyard_server())
+    print(vineyard_server)
+    sleep(2)
+    fuse_server = asyncio.run(start_fuse_server())
+    sleep(2)
+    print("server started")
+    client = connect_to_server()
+# test array
+    int_array =generate_array(Type.INT64)
+    test_fuse_array(int_array,client)
+    double_array = generate_array(Type.DOUBLE)
+    test_fuse_array(double_array,client)
+    
+    string_array = generate_array(Type.STRING)
+    test_fuse_string_array(string_array,client) 
+    print("array_test passed")
+# test df
+    int_df = generate_dataframe()
+    test_fuse_df(int_df,client)
+    print("df_test_passed")
+    interrupt_proc(fuse_server)
+    interrupt_proc(vineyard_server)
+
```

Comment:
Please format your python using isort and black.

You could add this file to the list in [setup.py](https://github.com/v6d-io/v6d/blob/main/setup.py#L114-L125)

Reply:
formatted and added